src/inject/dynamic-theme/style-manager.ts

```diff
@@ -318,50 +318,22 @@ export function manageStyle(element: StyleElement, {update, loadingStart, loadin
         return cssRules;
     }
 
-    let rulesChangeKey: number = null;
-    let rulesCheckFrameId: number = null;
-
-    function updateRulesChangeKey() {
-        const rules = safeGetSheetRules();
-        if (rules) {
-            rulesChangeKey = rules.length;
-        }
-    }
-
-    function didRulesKeyChange() {
-        const rules = safeGetSheetRules();
-        return rules && rules.length !== rulesChangeKey;
-    }
-
     function subscribeToSheetChanges() {
-        updateRulesChangeKey();
-        unsubscribeFromSheetChanges();
-        const checkForUpdate = () => {
-            if (didRulesKeyChange()) {
-                updateRulesChangeKey();
-                update();
-            }
-            rulesCheckFrameId = requestAnimationFrame(checkForUpdate);
-        };
-        checkForUpdate();
-    }
-
-    function unsubscribeFromSheetChanges() {
-        cancelAnimationFrame(rulesCheckFrameId);
+        element.addEventListener('updateSheet', update);
```

Comment:
There's a small chance of events name collision. Add some `darkreader` prefix to the event name.